### PR TITLE
Pull Request Checks

### PR DESCRIPTION
**Pull request by onetrickwolf**
Setting up GitHub action to run on pushes to main and on pull requests targeting main. Just runs the build to make sure it's not broken and does linting and prettier checks. Eventually we can add testing to this.

I think this is important to get in early so we don't have run away style issues.

- Added [pr-checks.yml](https://github.com/WarsWorld/WarsWorld/compare/main...onetrickwolf:WarsWorld:gh-action-checks#diff-1da7a27709d11ac16eef3de8e46ed10672b7b1d63a790ffbcbd23a94b19efdcb)
- Added [.prettierignore](https://github.com/WarsWorld/WarsWorld/compare/main...onetrickwolf:WarsWorld:gh-action-checks#diff-b640b344ee7f3f03d2a443795a5d0708ef50e2e6e34214109ab2aad13ad6ba98) so we aren't doing prettier checks on build folders etc
- Linting max warnings set to 10 while to give us some room while we are stabilizing the project
- [.nvmrc](https://github.com/WarsWorld/WarsWorld/compare/main...onetrickwolf:WarsWorld:gh-action-checks#diff-2c8effdf840690ccb88c7e21e56148978c6adb2c6926530c58ff0e47a9588b44) added to specify `node 18` for the project
- All other file changes are just prettier fixes to get build to initially pass

Tested running in my repo if you want to see: https://github.com/onetrickwolf/WarsWorld/pull/1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "start": "cross-env NODE_ENV=production node dist/server/prodServer.js",
     "studio": "prisma studio",
     "lint": "eslint --cache --ext \".js,.ts,.tsx\" --report-unused-disable-directives --report-unused-disable-directives src",
+    "prettier": "prettier --check .",
```
Comment (FunctionDJ) [inline package.json:29]: please change name to "lint-prettier" or similar to make obvious that this doesn't do any `--fix` stuff

Comment (FunctionDJ) [reply package.json:29]: actually, maybe just add this to `pnpm lint`

File: .gitignore [hunk 2]
```diff
@@ -13,8 +13,10 @@ tsconfig.tsbuildinfo
 
 # production
 /build
+/dist/
```
Comment (FunctionDJ) [inline .gitignore:16]: what tool generates /dist/?

File: .nvmrc
Comment (FunctionDJ) [inline .nvmrc:1]: i know about nvm but that's used individually and on the project-level pnpm checks the node version via package.json.
so i don't see the need for this config file.

**Review by FunctionDJ — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (onetrickwolf) [reply .gitignore:16]: `pnpm build` genetates a dist for me on main

**Review by megagreg72 — APPROVED** [review]
lgtm

**Comment by FemboyDeveloper** [comment]
LGTM as well, don't see a reason not to merge it with main already/

**Comment by onetrickwolf** [comment]
@FemboyDeveloper I am "You’re not [authorized](https://docs.github.com/articles/about-protected-branches/) to merge this pull request." but I may be blocked by @FunctionDJ review but yeah all set the merged whenever 👍

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,14 +31,18 @@ Running our project locally just takes 5 easy steps!
...
 ```
+
 Now go to localhost:3000 and you should be set! If you are having issues, feel free to contact someone in the team!
 
-``` Soon there will be a more newb/extensive setup your local environment guide here!```
+` Soon there will be a more newb/extensive setup your local environment guide here!`
+
 ## RoadMap / Vision
+
 Currently the project intends to recreate AWBW but with 2023 technology. Meaning we want to have a live league that works smoothly, a hot from the oven UI/UX thats reponsive and streamlines games, being able to look at multiple games at the same time, having a much faster website/not having lag, etc.
```
Comment (leekahung) [inline CONTRIBUTING.md:46]: Typo with "that's responsive" not "thats reponsive".

And what does it mean by "hot from the oven".

**Review by leekahung — COMMENTED** [review]
Minor stuff, see feedback.

File: CONTRIBUTING.md (hunk 3 above)
Comment (onetrickwolf) [reply CONTRIBUTING.md:46]: Nice catch but I didn't write this content, changes to [CONTRIBUTING.md](https://github.com/WarsWorld/WarsWorld/pull/91/files/c3dea4d1438f55f8a574fe716542e648b01326d1#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055) were only made due to prettier fixes to make the build pass. Perhaps best addressed in another PR?

Comment (leekahung) [reply CONTRIBUTING.md:46]: Yeah, I think we can do that.

**Review by FemboyDeveloper — APPROVED** [review]
Okay, I'm going to resolve the conflicts and merge to main.

**Pull request merged**